services/insights-handler/internal/handler/main.go

```diff
@@ -0,0 +1,722 @@
+	// default is 10 retry with 30 second delay = 5 minutes
+	err := try.Do(func(attempt int) (bool, error) {
+		var err error
+		messageQueue, err = mq.New(h.Config)
+		if err != nil {
+			log.Println(err,
+				fmt.Sprintf(
+					"Failed to initialize message queue manager, retrying in %d seconds, attempt %d/%d",
+					h.ConnectionRetryInterval,
+					attempt,
+					h.ConnectionAttempts,
+				),
+			)
+			time.Sleep(time.Duration(h.ConnectionRetryInterval) * time.Second)
+		}
+		return attempt < h.ConnectionAttempts, err
+	})
+	if err != nil {
+		log.Fatalf("Finally failed to initialize message queue manager: %v", err)
+	}
+	defer messageQueue.Close()
+
+	go func() {
+		for err := range messageQueue.Error() {
+			log.Println(fmt.Sprintf("Caught error from message queue: %v", err))
+		}
+	}()
+
+	forever := make(chan bool)
+
+	// Handle any tasks that go to the queue
+	log.Println("Listening for messages in queue lagoon-insights:items")
+	err = messageQueue.SetConsumerHandler("items-queue", processingIncomingMessageQueueFactory(h))
+	if err != nil {
+		log.Println(fmt.Sprintf("Failed to set handler to consumer `%s`: %v", "items-queue", err))
+	}
+	<-forever
+}
+
+type authedTransport struct {
+	wrapped http.RoundTripper
+	h       *Messaging
+}
+
+func (t *authedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	//grab events for project
+	token, err := jwt.OneMinuteAdminToken(t.h.LagoonAPI.TokenSigningKey, t.h.LagoonAPI.JWTAudience, t.h.LagoonAPI.JWTSubject, t.h.LagoonAPI.JWTIssuer)
+	if err != nil {
+		// the token wasn't generated
+		if t.h.EnableDebug {
+			log.Println(err)
+		}
+		return nil, err
+	}
+	req.Header.Set("Authorization", "bearer "+token)
+	return t.wrapped.RoundTrip(req)
+}
+
+func processingIncomingMessageQueueFactory(h *Messaging) func(mq.Message) {
+	return func(message mq.Message) {
+		var insights InsightsData
+		var resource ResourceDestination
+
+		incoming := &InsightsMessage{}
+		json.Unmarshal(message.Body(), incoming)
+
+		// Check labels for sbom data from message
+		if incoming.Labels != nil {
+			for key, value := range incoming.Labels {
+				if key == "lagoon.sh/insightsType" && value == "sbom" {
+					insights = InsightsData{
+						InputType:  Sbom,
+						LagoonType: Facts,
+					}
+				}
+				if key == "lagoon.sh/insightsType" && value == "sbom-gz" {
+					insights = InsightsData{
+						InputType:  SbomGz,
+						LagoonType: Facts,
+					}
+				}
+				if key == "lagoon.sh/insightsType" && value == "image-inspect-gz" {
+					insights = InsightsData{
+						InputType:  ImageInspectGz,
+						LagoonType: ImageInspectFacts,
+					}
+				}
+				if key == "lagoon.sh/insightsProject" {
+					resource.Project = value
+				}
+				if key == "lagoon.sh/insightsEnvironment" {
+					resource.Environment = value
+				}
+				if key == "lagoon.sh/insightsService" {
+					resource.Service = value
+				}
+				if key == "lagoon.sh/insightsFormat" {
```

Comment:
What's this label for?

Reply:
This was to allow control for different formats of insights data that we might get before we decided on gzipping it all. So if the type of insight data comes in as raw json or csv or whatever we can handle it differently.